aws-codeguruprofiler-profilinggroup/aws-codeguruprofiler-profilinggroup.json

```diff
@@ -16,6 +20,33 @@
       "maxLength": 255,
       "pattern": "^[\\w-]+$"
     },
+    "Permissions": {
+      "description": "The permissions attached for this profiling group.",
+      "type": "object",
+      "additionalProperties": false,
+      "required": [
+        "AgentPermissions"
+      ],
+      "properties": {
+        "AgentPermissions": {
+          "type": "object",
+          "description": "The permissions for the agent.",
+          "additionalProperties": false,
+          "required": [
+            "Principals"
+          ],
+          "properties": {
+            "Principals": {
+              "description": "The principals for the agent permissions.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/ArnIam"
+              }
+            }
+          }
+        }
```

Comment:
This structure seems quite deep:

```json
{
  "Type": "AWS::CodeGuruProfiler::ProfilingGroup",
  "Properties": {
    "ProfilingGroupName": "some-profiling-group",
    "Permissions": {
      "AgentPermissions": {
        "Principals": ["...list of principals... "]
      }
    }
  }
}
```

What do you think of making it a little more flatter by going with

```json
{
  "Type": "AWS::CodeGuruProfiler::ProfilingGroup",
  "Properties": {
    "ProfilingGroupName": "some-profiling-group",
    "Permissions": [
      {
        "ActionGroup": "agentPermissions",
        "Principals": ["...list of principals..."]
      }
    ]
  }
}
```

or

```json
{
  "Type": "AWS::CodeGuruProfiler::ProfilingGroup",
  "Properties": {
    "ProfilingGroupName": "some-profiling-group",
    "AgentPermissions": {
      "Principals": ["...list of principals... "]
    }
  }
}
```

?

Cloudformation is already quite verbose, so the more lightweight we can make it, the better :)

Reply:
Good point, I will update the code with having ```AgentPermissions``` as first-level property, thanks.